### PR TITLE
[spark-operator]: Fix annotations

### PR DESCRIPTION
**Pull request by alexpirogovski**
### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove irrelevant fields.]
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[mychartname]`)

### Description:
Remove annotations from the resources so that the hook (and caBundle) is created only once in service's lifetime

### CODE REVIEW

File: stable/spark-operator/templates/rbac.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
   name: {{ include "spark-operator.fullname" . }}-cr
+  annotations:
```
Comment (liranbg) [inline stable/spark-operator/templates/rbac.yaml]: 👀  why

File: stable/spark-operator/templates/rbac.yaml [hunk 2]
```diff
@@ -25,6 +26,7 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRoleBinding
 metadata:
   name: {{ include "spark-operator.fullname" . }}-crb
+  annotations:
```
Comment (liranbg) [inline stable/spark-operator/templates/rbac.yaml]: no need to declare if empty / no default values / no helm values

File: stable/spark-operator/templates/webhook-cleanup-job.yaml [hunk 3]
```diff
@@ -6,7 +6,6 @@ metadata:
   annotations:
     "helm.sh/hook": pre-delete
     "helm.sh/hook-delete-policy": hook-succeeded
```
Comment (liranbg) [inline stable/spark-operator/templates/webhook-cleanup-job.yaml]: remove these instead

File: stable/spark-operator/templates/serviceaccount.yaml [hunk 4]
```diff
@@ -4,8 +4,6 @@ kind: ServiceAccount
 metadata:
   name: {{ include "spark-operator.serviceAccountName" . }}
   annotations:
-    "helm.sh/hook": pre-install
-    "helm.sh/hook-delete-policy": hook-failed, before-hook-creation
```
Comment (liranbg) [inline stable/spark-operator/templates/serviceaccount.yaml:8]: 👏🏼

File: stable/spark-operator/templates/webhook-cleanup-job.yaml [hunk 5]
```diff
@@ -6,7 +6,6 @@ metadata:
...
-    {{- toYaml .Values.webhook.cleanupAnnotations | nindent 4 }}
```
Comment (liranbg) [inline stable/spark-operator/templates/webhook-cleanup-job.yaml]: undo

File: stable/spark-operator/values.yaml [hunk 6]
```diff
@@ -99,12 +99,8 @@ webhook:
   namespaceSelector: ""
   # -- The annotations applied to init job, required to restore certs deleted by the cleanup job during upgrade
   initAnnotations:
-    "helm.sh/hook": pre-install, pre-upgrade
-    "helm.sh/hook-weight": "50"
   # -- The annotations applied to the cleanup job, required for helm lifecycle hooks
   cleanupAnnotations:
-    "helm.sh/hook": pre-delete, pre-upgrade
-    "helm.sh/hook-delete-policy": hook-succeeded
```
Comment (liranbg) [inline stable/spark-operator/values.yaml]: imo leave here, next time we will fork the spark operator helm chart, it won't conflict or raise a question

File: stable/spark-operator/values.yaml [hunk 7]
```diff
@@ -99,12 +99,8 @@ webhook:
   namespaceSelector: ""
   # -- The annotations applied to init job, required to restore certs deleted by the cleanup job during upgrade
   initAnnotations:
-    "helm.sh/hook": pre-install, pre-upgrade
-    "helm.sh/hook-weight": "50"
```
Comment (liranbg) [inline stable/spark-operator/values.yaml]: same comment as below

**Review by liranbg — CHANGES_REQUESTED** [review]
See comments below

File: stable/spark-operator/templates/webhook-cleanup-job.yaml (hunk 3 above)
Comment (alexpirogovski) [reply stable/spark-operator/templates/webhook-cleanup-job.yaml]: These were in the previous chart too, I guess they kick in when the chart is uninstalled

File: stable/spark-operator/templates/webhook-cleanup-job.yaml (hunk 5 above)
Comment (alexpirogovski) [reply stable/spark-operator/templates/webhook-cleanup-job.yaml]: It breaks yaml if not removed

File: stable/spark-operator/templates/serviceaccount.yaml (hunk 4 above)
Comment (alexpirogovski) [reply stable/spark-operator/templates/serviceaccount.yaml:8]: Could you please use words? I don't speak middle-schoolish

File: stable/spark-operator/values.yaml (hunk 7 above)
Comment (alexpirogovski) [reply stable/spark-operator/values.yaml]: Done

File: stable/spark-operator/values.yaml (hunk 6 above)
Comment (alexpirogovski) [reply stable/spark-operator/values.yaml]: Done

File: stable/spark-operator/templates/webhook-cleanup-job.yaml (hunk 5 above)
Comment (alexpirogovski) [reply stable/spark-operator/templates/webhook-cleanup-job.yaml]: Done

File: stable/spark-operator/templates/rbac.yaml (hunk 2 above)
Comment (alexpirogovski) [reply stable/spark-operator/templates/rbac.yaml]: Done

**Pull request merged**